sys/net/gnrc/link_layer/gnrc_mac/internal.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net_gnrc_mac
+ * @file
+ * @brief       Internal functions of GNRC_WMAC
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#include <stdbool.h>
+#include <net/gnrc.h>
+#include <net/gnrc/mac/internal.h>
+#include <net/gnrc/mac/types.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if ((GNRC_MAC_TX_QUEUE_SIZE != 0)||(GNRC_MAC_RX_QUEUE_SIZE != 0))
+gnrc_priority_pktqueue_node_t* _alloc_pktqueue_node(gnrc_priority_pktqueue_node_t* nodes[], uint32_t size)
+{
+    /* search for free packet_queue_node */
+    for (size_t i = 0; i < size; i++) {
+        if((nodes[i]->pkt == NULL) &&
+           (nodes[i]->next == NULL)) {
+            return nodes[i];
+        }
+    }
+
+    return NULL;
+}
+#endif
+/* endif for `#if((GNRC_MAC_TX_QUEUE_SIZE != 0)||(GNRC_MAC_RX_QUEUE_SIZE != 0))` */
+
+#if GNRC_MAC_TX_QUEUE_SIZE != 0
+#if GNRC_MAC_NEIGHBOR_COUNT != 0
+int _find_neighbor(gnrc_mac_tx_t* tx, uint8_t* dst_addr, int addr_len)
```

Comment:
@miri64 Sorry for being nagging here. And sorry I didn't get the point about "mainly for compiler-based optimization" due to my weak knowledge~  How is realized? should I `const uint8_t* dst_addr` or`uint8_t* const dst_addr`? My idea is to get the point clear such that next time I can avoid it to save reviewer's time. ;-)